### PR TITLE
Support spellchecking in markdown & text mode

### PR DESCRIPTION
**Pull request by khazhoyan**
re #110

### CODE REVIEW

File: lib/ace/mode/markdown_worker.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+define(function(require, exports, module) {
```
Comment (dbarashev) [inline lib/ace/mode/markdown_worker.js]: add copyright, author and license notice (similar to how it is done in ACE code)

File: lib/ace/mode/markdown_worker.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    this.mySpellingCheckOptions = {};
+    this.myTypos = {};
+    var highlighter = new MarkdownHighlightRules();
+    this.myBgTokenizer = new BackgroundTokenizer(new Tokenizer(highlighter.getRules()));
+    this.myBgTokenizer.setDocument(this.doc);
+    this.myBgTokenizer.start(0);
+    this.myParser = new TextParser(this.myBgTokenizer);
+};
+
+oop.inherits(MarkdownWorker, Mirror);
```
Comment (dbarashev) [inline lib/ace/mode/markdown_worker.js]: maybe we can extract common code into PapeeriaWorker and extends both LaTeX and Markdown from that worker?

File: lib/ace/mode/text_worker.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    this.mySpellingCheckOptions = {};
+    this.myTypos = {};
+    var highlighter = new TextHighlightRules();
+    this.myBgTokenizer = new BackgroundTokenizer(new Tokenizer(highlighter.getRules()));
+    this.myBgTokenizer.setDocument(this.doc);
+    this.myBgTokenizer.start(0);
+    this.myParser = new TextParser(this.myBgTokenizer);
+};
+
+oop.inherits(TextWorker, Mirror);
```
Comment (dbarashev) [inline lib/ace/mode/text_worker.js]: The same about inheriting from the common worker

File: lib/ace/mode/text_parser.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
+define(function(require, exports, module) {
```
Comment (dbarashev) [inline lib/ace/mode/text_parser.js]: copyright, authot and license note pls

File: lib/ace/mode/text_worker.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
+define(function(require, exports, module) {
```
Comment (dbarashev) [inline lib/ace/mode/text_worker.js]: copyright, author and license note

File: lib/ace/mode/markdown_worker.js (hunk 2 above)
Comment (khazhoyan) [inline lib/ace/mode/markdown_worker.js]: good call, done

File: lib/ace/mode/markdown_worker.js (hunk 1 above)
Comment (khazhoyan) [inline lib/ace/mode/markdown_worker.js]: done

**Pull request merged**